src/packages/admin-ui-components/src/detail-panel/fields/markdown-field/menu-bar/list-options.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useMemo, useState } from 'react';
+import {
+	EditorOrderedListIcon,
+	EditorBlockquoteIcon,
+	EditorUnOrderedListIcon,
+} from '../../../../assets';
+import { SectionProps } from './utils';
+import styles from './styles.module.css';
+
+export const ListOptions = (props: SectionProps) => {
+	const { editor, options } = props;
+	const [showItems, setShowItems] = useState(false);
+
+	const getMainAndOptionButtons = useCallback(() => {
+		const handleBulletListClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+			event.preventDefault();
+			event.stopPropagation();
+			editor.chain().focus().toggleBulletList().run();
+		};
+
+		const handleOrderedListClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+			event.preventDefault();
+			event.stopPropagation();
+			editor.chain().focus().toggleOrderedList().run();
+		};
+
+		const handleBlockquoteClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+			event.preventDefault();
+			event.stopPropagation();
+			editor.chain().focus().toggleBlockquote().run();
+		};
```

Comment:
This pattern is quite common. Would it make sense to have a generator function util something like:

```
const editorClickHandler = (functionName: keyof typeof editor) => (event: React.MouseEvent<HTMLButtonElement>) => {
     event.preventDefault();
     event.stopPropagation();
     editor.chain().focus()[functionName]().run();
};
```

Or perhaps it's `<T extends keyof typeof editor>` or similar, the type being, "Some function with no parameters that exists on the result of `.focus()` that I can call.

Then we can just do:
```
onClick={editorClickHandler('toggleBulletList')}
```
everywhere with just changing that string.